app/src/main/java/com/example/cactusnotes/SignUpActivity.kt

```diff
@@ -4,12 +4,31 @@ import android.os.Bundle
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import com.example.cactusnotes.databinding.SignUpActivityBinding
+import com.example.cactusnotes.model.CactusModel
+import com.example.cactusnotes.model.RegisterErrorResponse
+import com.example.cactusnotes.model.RegisterResponse
+import com.example.cactusnotes.service.CactusApi
 import com.example.cactusnotes.validations.EmailValidator
 import com.example.cactusnotes.validations.PasswordValidator
 import com.example.cactusnotes.validations.UsernameValidator
+import com.google.android.material.snackbar.Snackbar
+import com.google.gson.GsonBuilder
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
 
 class SignUpActivity : AppCompatActivity() {
     private lateinit var binding: SignUpActivityBinding
+
+    val api = Retrofit.Builder()
+        .baseUrl("https://apps.cactus.school")
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+        .create(CactusApi::class.java)
```

Comment:
You should move the api object elsewhere, since other Activities will also need to reach this object.